### PR TITLE
1117 Add TorchScript compatible test

### LINKED ISSUE #1117: Add TorchScript compatible test for all the networks
**Is your feature request related to a problem? Please describe.**
In order to make sure all the networks can be exported and loaded in `TorchScript`, need to add more unit tests for them.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #1117 .

### Description
This PR added TorchScript tests for several networks, other networks contain some logic that can't support `torch.jit.script` and not easy to fix.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh --codeformat --coverage`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Review by ericspod — COMMENTED** [review]
It looks good other than the comment I made. I also stated in the related issue what needs to be done for the PSP class, that is the `up8/16/32/64` members need to be always present so can be `nn.Identity` if not needed, and line 303 of ahnet.py can be `interpolate_size = x.shape[2:]`. I was able to get Torchscript to work with AHNet with these changes.

File: tests/utils.py [hunk 1]
```diff
@@ -96,6 +96,18 @@ def expect_failure_if_no_gpu(test):
...
         return test
 
+def test_script_save(net, inputs):
+    scripted = torch.jit.script(net)
+    buffer = scripted.save_to_buffer()
+    reloaded_net = torch.jit.load(BytesIO(buffer))
+    net.eval()
+    reloaded_net.eval()
+    with torch.no_grad():
+        result1 = net.forward(inputs)
```
[the remark this answers is not in the document]
Comment (Nic-Ma) [reply tests/utils.py]: updated in the latest commit.
Thanks.

**Comment by Nic-Ma** [comment]
Hi @ericspod ,

Thanks very much for your review.
I have updated the PR according to your comments, now AHNet works, could you please help review it again?
There are some other networks that are not covered in this PR, because they still have different kinds of TorchScript errors that I didn't find quick fix, maybe we can try to investigate them one by one later?

Thanks.

**Review by ericspod — APPROVED** [review]
We can review the other networks later, I suspect some of the same issues are present in them as we saw here. Some of the reference networks meant to re-implement published work might not be compatible ever.

**Pull request merged**